src/main/java/org/opensrp/web/rest/PlanResource.java

```diff
@@ -279,6 +287,30 @@ public void setLocationService(PhysicalLocationService locationService) {
 
 	}
 
+	/**
+	 * This methods provides an API endpoint that searches for all plan Identifiers
+	 *
+	 * @return A list of plan Identifiers
+	 */
+	@RequestMapping(value = "/findIds", method = RequestMethod.GET, produces = {
+			MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<Identifier> findIds(@RequestParam(value = SERVER_VERSIOIN, required = false)  long serverVersion,
+										  @RequestParam(value = IS_DELETED, defaultValue = FALSE, required = false ) boolean isDeleted) {
+
+		try {
+			Pair<List<String>, Long> planIdsPair = planService.findAllIds(serverVersion, DEFAULT_GET_ALL_IDS_LIMIT, isDeleted);
+			Identifier identifiers = new Identifier();
+			identifiers.setIdentifiers(planIdsPair.getLeft());
+			identifiers.setLastServerVersion(planIdsPair.getRight());
+
+			return new ResponseEntity<>(identifiers, HttpStatus.OK);
+		} catch (Exception e) {
```

Comment:
can we can catch something more specific?

Reply:
This catches all possible errors preventing leaking of stack-trace. This can be removed the global error handler is configured https://github.com/OpenSRP/opensrp-server-web/issues/245